src/rebuild/scan.c

```diff
@@ -899,7 +899,10 @@ rebuild_tgt_scan_handler(crt_rpc_t *rpc)
 	}
 
 	dss_rpc_reply(rpc, DAOS_REBUILD_DROP_SCAN);
-	d_rank_list_free(fail_list);
+	/* will fix cart to call co_post_reply() for this case, freeing
+	 * it immediately at here is potentially unsafe.
+	 */
+	/* d_rank_list_free(fail_list); */
```

Comment:
for normal RPC, it is safe to free it just after crt_reply_send() returns, because it already be serialized/packed so safe to free.
For corpc's handler, it is unsafe in that case because cart can internally queue it for reply aggregation and have not been serialized. 
currently cart provide co_post_reply callback but it lost some cases that will cause some memleak, I'll fix it in cart later. But in DAOS it needs to depend on co_post_reply to free it.

Reply:
I see, thanks for the explanation. Seems disallowing CRT_PTR in corpc reply could make things simpler.

Reply:
Yes, it looks like a easy-to-make mistake